src/api/procedures/addInstruction.ts

```diff
@@ -301,15 +339,15 @@ export async function getAuthorization(
         ? TxTags.settlement.AddAndAffirmInstruction
         : TxTags.settlement.AddInstruction
     );
-    portfolios.push(...portfoliosList);
+    portfolios = unionWith(portfolios, portfoliosList, isEqual);
   });
 
   return {
     identityRoles: [{ type: RoleType.VenueOwner, venueId }],
     signerPermissions: {
       tokens: [],
-      portfolios: uniq(portfolios),
-      transactions: uniq(transactions),
+      portfolios,
+      transactions: union(transactions),
```

Comment:
The idea was to use union instead of pushing the transaction tags into the array. Using it here doesn't make sense.

```
transactions = union(transactions, [portfoliosList.length ? TxTags.settlement.AddAndAffirmInstruction : TxTags.settlement.AddInstruction]);
```